builder/parser/line_parsers.go

```diff
@@ -195,6 +192,44 @@ func parseLabel(rest string) (*Node, map[string]bool, error) {
 	return parseNameVal(rest, "LABEL")
 }
 
+// parses a statement with multiple keywords definition and name-value
+// assignment, like `name1= name2=value`.
+// Note that this is a stricter format than the old format of assignment, in a way
+// that it only allows assignment of the form `name=value`. And that a name
+// definition or assignment to an empty value is designated by `name=`.
+func parseNameOrNameVal(rest string) (*Node, map[string]bool, error) {
+	words := parseWords(rest)
+	if len(words) == 0 {
+		return nil, nil, nil
+	}
+
+	var (
+		rootnode *Node
+		prevNode *Node
+	)
+	for i, word := range words {
+		if !strings.Contains(word, "=") {
+			return nil, nil, fmt.Errorf("Syntax error - can't find = in %q. Must be of the form: name=value", word)
```

Comment:
If we just parse the line as a list of space separated words then arg() can check for more than one (if we want to ban more than one) and check each word for "=".  Yes it can be viewed as moving parsing logic into arg() but I don't think of this as parsing logic as much as I view it as "semantic checking".  I know not all of the parsing code does it this way, but part of that is due to historical reasons and not necessarily good design.